src/parser/parser-office.ee.ts

```diff
@@ -0,0 +1,146 @@
+import { XMLParser } from 'fast-xml-parser';
+
+/**
+ * Parse a document file and extract text content
+ * @param documents All required documents
+ * @returns Extracted text as a string
+ */
+export async function extractOfficeDocument(documents: {
+  [key: string]: Uint8Array;
+}): Promise<string> {
+  const parser = new XMLParser({
+    alwaysCreateTextNode: true,
+    ignoreAttributes: true,
+    preserveOrder: true,
+    removeNSPrefix: true,
+  });
+
+  let combinedText = '';
+
+  for (const [, content] of Object.entries(documents)) {
+    const parsed = parser.parse(
+      new TextDecoder('utf8').decode(content),
+    ) as XMLNode;
+
+    const extractedText = await recursiveXMLParsedExtract({
+      source: parsed,
+      text: '',
+    });
+
+    combinedText += `${formatText(extractedText)}\n`;
+  }
+
+  return combinedText.trim();
+}
+
+/**
+ * This function takes an unknown source, and will recursively try to extract text from it.
+ * Handles paragraphs and headers separately to preserve newlines.
+ * @param root named params
+ * @param root.source an object or a list to extract text from
+ * @param root.text a string to concatenate with the result of this function. Set to undefined to create a new string
+ * @returns either undefined or a string concatenated with root.text
+ */
+async function recursiveXMLParsedExtract({
+  source,
+  text,
+}: {
+  source: XMLNode;
+  text: string | undefined;
+}): Promise<string | undefined> {
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition, sonarjs/different-types-comparison
+  if (typeof source === 'object' && source !== null && !Array.isArray(source)) {
+    // the xml parser will indicate text with #text, so we only need to seek those tags and concat them
+    if (
+      '#text' in source &&
+      typeof source['#text'] === 'string' &&
+      source['#text'].length > 0
+    ) {
+      const cleanedText = source['#text'];
+      return text ? `${text} ${cleanedText}` : cleanedText;
```

Comment:
You can simplify this expression if you are using a for loop

```ts
return text + source['#text'];
```

and in your function parameters:

```ts
text: string;
```

and your function return, since now you are editing the original string instead of building a new one

```ts
: Promise<void>
```